src/api/rate_limiter.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+import requests
+import uuid
+
+from datetime import datetime
+
+
+class RequestEntry(object) :
+    """
+    Class for request object to put in the queue for the rate limiter
+    """
+    def __init__(self,url,cookies,key,method) -> None:
+        self.url = url
+        self.cookies = cookies
+        self.key = key
+        self.method = method
+
+class RateLimiter() :
+    """
+    Class that takes care of sending the requests at a maximum rate (25/sec) 
+        and giving back the result to the calling functions when it is received (async probably)
+    """
+    def __init__(self) :
+        self.requests = {}
+        self.queue = asyncio.Queue()
+        asyncio.create_task(self.handle_requests())
+
+
+    async def handle_requests(self) :
+        print(f"Starting rate_limiter task...")
+        while True :
```

Comment:
I guess thanks to the async this doesn't create an infinite loop?

Reply:
Yes this creates an infinite task, but as soon as this asynchronous task is awaiting something either `await self.queue.get()` or the `asyncio.sleep()`, it is not actually being executed.

Reply:
Ok it's great actually! tks for the explanation